src/lib.rs

```diff
@@ -16,6 +16,74 @@
+//!   identical set or a subset of the original's permissions. Permission
+//!   escalation by untrusted processes is impossible.
+//!
+//! Here's a box diagram of two processes who each have a mailbox and a
+//! capability to the other's mailbox to help explain the mental model of a
+//! full Flue-based actor system:
+//!
+//! ```
+//!         Process A               Post Office               Process B
+//! ┌───────────────────────┐     ┌─────────────┐     ┌───────────────────────┐
+//! │         Table         │     │             │     │         Table         │
+//! │ ┌───────────────────┐ │     │ ┌─────────┐ │     │ ┌───────────────────┐ │
+//! │ │                   │ │     │ │         │ │     │ │                   │ │
+//! │ │ ┌───────────────┐ │ │  ┌──┼─► Route B ├─┼───┐ │ │ ┌───────────────┐ │ │
+//! │ │ │ Capability A  ├─┼─┼──┘  │ │         │ │   │ │ │ │ Capability B  ├─┼─┼─┐
+//! │ │ └───────────────┘ │ │     │ └─────────┘ │   │ │ │ └───────────────┘ │ │ │
+//! │ │                   │ │     │             │   │ │ │                   │ │ │
+//! │ └────────▲──────────┘ │     │ ┌─────────┐ │   │ │ └────────▲──────────┘ │ │
+//! │          │            │     │ │         │ │   │ │          │            │ │
+//! │   ┌──────┴───────┐    │  ┌──┼─┤ Route A ◄─┼─┐ │ │   ┌──────┴───────┐    │ │
+//! │   │   Mailbox A  ◄────┼──┘  │ │         │ │ │ └─┼───►   Mailbox B  │    │ │
+//! │   └──────────────┘    │     │ └─────────┘ │ │   │   └──────────────┘    │ │
+//! │                       │     │             │ │   │                       │ │
+//! └───────────────────────┘     └─────────────┘ │   └───────────────────────┘ │
+//!                                               │                             │
+//!                                               └─────────────────────────────┘
+//! ```
+//!
+//! Both processes share a post office, and mailboxes A and B have associated
+//! routes A and B in that mailbox. Capability A belongs in process A's table
+//! and references route B, and capability B lives in process B's table and
+//! references route A. Mailboxes A and B reference their associated process's
+//! table so that when they receive capabilities, they can insert those
+//! capabilities into their process's table.
+//!
+//! To send a message signal from process A to process B using capability A,
+//! Flue performs the following steps on that message:
+//! 1. Process A's table confirms that capability A does indeed have the
+//!    permission to send messages.
+//! 2. Process A's table sends the message to route B's address in the post
+//!    office.
+//! 3. The post office looks up route B by address, finds mailbox B's zero-copy
+//!    message channel, and uses it to send the message.
+//! 4. The message waits in mailbox B's queue until mailbox B receives it.
+//! 5. Mailbox B processes the message and inserts any capabilities inside of
+//!    it into process B's mailbox.
```

Comment:
Absolutely amazing. Which tool did you use for it, because I will have to use this now as well :fire:

Reply:
I used [this](https://asciiflow.com).